tasks/Task.cpp

```diff
@@ -45,42 +48,40 @@ struct controldev_websocket::JoystickHandler : WebSocket::Handler {
     }
 };
 
+Json::Value jdata;
+Json::CharReaderBuilder rbuilder;
+std::unique_ptr<Json::CharReader> const reader(rbuilder.newCharReader());
 bool Task::updateRawCommand(const char *data){
-    Json::Value jdata;
-    Json::CharReaderBuilder rbuilder;
-    std::unique_ptr<Json::CharReader> const reader(rbuilder.newCharReader());
+    ++received;
     std::string errs;
 
     if (!reader->parse(data, data+std::strlen(data), &jdata, &errs)){
-        std::cout << errs << std::endl;
+        LOG_ERROR_S << "Failed parsing the message, got error: " << errs << std::endl;
+        ++errors;
         return false;
     }
-    if (!(jdata.isMember("status") && jdata.isMember("axes") && jdata.isMember("buttons"))) {
+    if (!jdata.isMember("status") || !jdata.isMember("axes") || !jdata.isMember("buttons")) {
+        LOG_ERROR_S << "Invalid message, it does not contain required fields." << std::endl;
+        ++errors;
```

Comment:
Do the statistics 9both received and errors) in the caller of `updateRawCommand`, which will guarantee that we won't forget one error path.